src/clj/rems/api/services/licenses.clj

```diff
@@ -0,0 +1,79 @@
+(ns rems.api.services.licenses
+  "Serving licenses for API."
+  (:require [rems.common-util :refer [distinct-by]]
+            [rems.db.core :as db]
+            [rems.db.licenses :as licenses]
+            [rems.db.resource :as resource]
+            [rems.db.workflow :as workflow])
+  (:import (java.io FileInputStream ByteArrayOutputStream)))
+
+(defn create-license! [{:keys [title licensetype textcontent localizations attachment-id]} user-id]
```

Comment:
this feels like it should be in rems.db.licences

Reply:
Hmm, one problem is that in the current implementation, the functions in db.licenses have wrappers in api.services.licenses so that api.licenses can depend only on api.services.licenses and not on db.licenses directly.

I could change api.licenses to depend also on db.licenses, but I'm not sure what's the logic from api.licenses viewpoint behind figuring out which functions belongs where -- it feels arbitrary. Having wrappers here makes it at least consistent.

But moving (most of) these functions to db.licenses makes this intermediate step mostly just a collection of wrappers.

One criterion would be to have a minimal set of functions in db.licenses so that the other services have the functions they need, but not more. It is how it's done in the current implementation -- I'll need get-licenses in other services, but not create-license!.

Reply:
Ok, that'll work I guess